src/main/java/org/wildfly/security/credential/store/impl/MaskedPasswordStore.java

```diff
@@ -0,0 +1,174 @@
+
+import javax.crypto.Cipher;
+import javax.crypto.SecretKey;
+import javax.crypto.SecretKeyFactory;
+import javax.crypto.spec.PBEKeySpec;
+import javax.crypto.spec.PBEParameterSpec;
+
+import org.wildfly.security.credential.store.CredentialStoreSpi;
+import org.wildfly.security.util.Alphabet;
+import org.wildfly.security.util.ByteIterator;
+
+/**
+ * Pseudo credential store which is able to get credential from masked password string.
+ * It parameters attributes passed to {@link #initialize(Map)} for getting additional information to decode the password.
+ *
+ * @author <a href="mailto:pskopek@redhat.com">Peter Skopek</a>
+ */
+public class MaskedPasswordStore extends CommandCredentialStore {
+
+    /**
+     * Parameter name which denotes salt
+     */
+    public static final String SALT = "salt";
+    /**
+     * Parameter name which denotes iteration count
+     */
+    public static final String ITERATION_COUNT = "iteration";
+    /**
+     * Parameter name which denotes PBE algorithm
+     */
+    public static final String PBE_ALGORITHM = "algorithm";
+    /**
+     * Parameter name which denotes initial key for PBE algorithm
+     */
+    public static final String INITIAL_KEY = "initialKey";
+
+    /**
+     * Prefix to used when masked password needs to be detected.
+     */
+    public static String PASS_MASK_PREFIX = "MASK-";
+
+    private static final char[] DEFAULT_PBE_KEY = "somearbitrarycrazystringthatdoesnotmatter".toCharArray();
+    static final String DEFAULT_PBE_ALGORITHM = "PBEwithMD5andDES";
+
+
+    /**
+     * Type of {@link CredentialStoreSpi} implementation. Will be used as algorithm name when registering service in
+     * {@link org.wildfly.security.WildFlyElytronProvider}.
+     */
+    public static final String MASKED_PASSWORD_STORE = "MaskedPasswordStore";
+
+    /**
+     * Default constructor.
+     */
+    public MaskedPasswordStore() {
+        storeName = "masked";
+    }
+
+    /**
+     * Executes command in operating system using {@link Runtime#exec(String)} method. Grabs the output and return it for further processing.
+     * In case of Java Security Manager active uses doPrivileged to start the command.
+     * @param passwordCommand command as operating system accepts
+     * @return output from the {@link Process} resulting of command execution
+     * @throws Throwable when something goes wrong
+     */
+    @Override
+    char[] executePasswordCommand(String passwordCommand) throws Throwable {
+        String secret;
+        if (passwordCommand.startsWith(PASS_MASK_PREFIX)) {
+            secret = passwordCommand.substring(PASS_MASK_PREFIX.length());
+        } else {
+            secret = passwordCommand;
+        }
+        return decode(secret, getSalt(), getIterationCount(), getPbeAlgorithm(), getInitialKey());
+    }
+
+    /**
+     * This decodes password encoded using {@code PBEUtils} class of PicketBox.
+     *
+     * Method ported from PicketBox to maintain backward compatibility.
```

Comment:
First line of the comment can probably be updated as you have ported the method instead of calling the utils class.